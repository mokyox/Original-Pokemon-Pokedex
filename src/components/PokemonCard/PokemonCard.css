.img {
  padding: 30px;
  margin: 10px;
  text-align: center !important;
}

.description {
  font-size: 0.85em;
}

.card-container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: stretch;
      -ms-flex-align: stretch;
          align-items: stretch;
}

.pokemon-name,
.pokemon-stat-title,
.pokemon-stats {
  font-weight: 700;
  text-transform: capitalize;
}

.pokemon-stats {
  font-size: 1.5em;
  margin-bottom: 15px;
}

.pokemon-stat-title {
  padding: 5px 0px;
}

.pokemon-stat-title span {
  font-weight: 400;
}
