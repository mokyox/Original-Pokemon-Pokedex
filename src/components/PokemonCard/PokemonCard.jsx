import React from "react";
import Link from "next/link";
import {
  PokemonTypeBadgeContainer,
  PokemonTypeBadge,
  PokemonName,
  PokemonNumber,
  PokemonCardContainer,
} from "./styles";
import {
  LazyLoadImage,
  trackWindowScroll,
} from "react-lazy-load-image-component";

const PokemonCard = ({ name, id, image, type, scrollPosition }) => {
  return (
    <PokemonCardContainer data-testid="pokemon-card">
      <Link href={`/pokemon/[id]`} as={`/pokemon/${id}`}>
        <a>
          <PokemonName>{name}</PokemonName>
        </a>
      </Link>
      <PokemonNumber>#{id}</PokemonNumber>
      <PokemonTypeBadgeContainer>
        {type.map((type) => (
          <PokemonTypeBadge key={type} type={type}>
            {type}
          </PokemonTypeBadge>
        ))}
      </PokemonTypeBadgeContainer>

      <div>
        <LazyLoadImage
          src={image}
          alt={`pokemon: ${name}`}
          height="96px"
          width="96px"
          effect="blur"
          scrollPosition={scrollPosition}
        />
      </div>
    </PokemonCardContainer>
  );
};

export default trackWindowScroll(PokemonCard);
