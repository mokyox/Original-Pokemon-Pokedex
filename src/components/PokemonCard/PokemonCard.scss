.img {
  padding: 30px;
  margin: 10px;
  text-align: center !important;
}

.description {
  font-size: 0.85em;
}

.card-container {
  display: flex;
  align-items: stretch;
}

.pokemon-name,
.pokemon-stat-title,
.pokemon-stats {
  font-weight: 700;
  text-transform: capitalize;
}

.pokemon-stats {
  font-size: 1.5em;
  margin-bottom: 15px;
}

.pokemon-stat-title {
  padding: 5px 0px;
  span {
    font-weight: 400;
  }
}
