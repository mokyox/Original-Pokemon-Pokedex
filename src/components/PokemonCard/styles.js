import styled from "styled-components";
import { colors } from "../utils";

export const PokemonCard = styled.div``;

export const PokemonName = styled.h1`
  font-size: 1.5rem;
  font-weight: 700;
`;

export const PokemonNumber = styled.p`
  font-size: 1rem;
  font-weight: 500;
`;

export const PokemonTypeBadge = styled.span`
  background-color: ${({ type }) => colors[type]};
`;
