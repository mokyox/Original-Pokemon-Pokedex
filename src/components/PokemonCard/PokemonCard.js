import React from "react";
import { PokemonCardStyle } from "./styles";
import { colors } from "../utils";

const PokemonCard = ({
    name,
    id,
    image,
    hp,
    attack,
    defense,
    specialAttack,
    specialDefense,
    speed,
    type,
    description,
    ability,
}) => {
    return (
        <>
            <PokemonCardStyle />
            <div className="card-container col-lg-3 col-md-4 col-sm-6 col-12 mb-3">
                <div className="card bg-dark text-white">
                    <div className="text-nowrap p-1">
                        <h4 className="pokemon-name">{name}</h4>
                        <p>#{id}</p>
                        {type.map((type) => (
                            <span
                                style={{
                                    backgroundColor: `${colors[type]}`,
                                }}
                                className="p-1 m-1 badge"
                                key={type}
                            >
                                {type}
                            </span>
                        ))}
                    </div>
                    <div className="card-body">
                        <img
                            src={image}
                            alt={`pokemon: ${name}`}
                            loading="lazy"
                            height="96px"
                            width="96px"
                        />
                        <div className="card-title text-left">
                            <h5 className="pokemon-stats">Stats</h5>
                            <h6 className="pokemon-stat-title">
                                Ability: <span>{ability}</span>
                            </h6>
                            <p className="pokemon-stat-title">
                                HP: <span>{hp}</span>
                            </p>
                            <p className="pokemon-stat-title">
                                Attack: <span>{attack}</span>
                            </p>{" "}
                            <p className="pokemon-stat-title">
                                Defense: <span>{defense}</span>
                            </p>
                            <p className="pokemon-stat-title">
                                Special Attack: <span>{specialAttack}</span>
                            </p>{" "}
                            <p className="pokemon-stat-title">
                                Special Defense: <span>{specialDefense}</span>
                            </p>{" "}
                            <p className="pokemon-stat-title">
                                Speed: <span>{speed}</span>
                            </p>
                            <p className="pokemon-stat-title">Description:</p>
                            <span className="description">{description}</span>
                        </div>
                    </div>
                </div>
            </div>
        </>
    );
};

export default PokemonCard;
