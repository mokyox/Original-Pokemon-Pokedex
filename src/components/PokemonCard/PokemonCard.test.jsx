import React from "react";
import PokemonCard from "./PokemonCard";
import renderer from "react-test-renderer";

describe("<PokemonCard/>", () => {
    const pokemonInfo = {
        id: 1,
        name: "Bulbasaur",
        image: "",
        type: ["Grass", "Poison"],
    };
    it("renders correctly", () => {
        const tree = renderer.create(<PokemonCard {...pokemonInfo} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
