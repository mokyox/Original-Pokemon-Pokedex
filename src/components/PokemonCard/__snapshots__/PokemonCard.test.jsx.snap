// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PokemonCard/> renders correctly 1`] = `
<div
  className="styles__PokemonCardContainer-sc-1rlmv0d-0 ljUTdH"
  data-testid="pokemon-card"
>
  <a
    href="/pokemon/1"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <h2
      className="styles__PokemonName-sc-1rlmv0d-1 jqiFDT"
    >
      Bulbasaur
    </h2>
  </a>
  <p
    className="styles__PokemonNumber-sc-1rlmv0d-2 iUnciT"
  >
    #
    1
  </p>
  <div
    className="styles__PokemonTypeBadgeContainer-sc-1rlmv0d-3 blForx"
  >
    <span
      className="styles__PokemonTypeBadge-sc-1rlmv0d-4 kQwjYZ"
      type="Grass"
    >
      Grass
    </span>
    <span
      className="styles__PokemonTypeBadge-sc-1rlmv0d-4 iCowM"
      type="Poison"
    >
      Poison
    </span>
  </div>
  <div>
    <span
      className=" lazy-load-image-background blur"
      style={
        Object {
          "backgroundImage": "",
          "backgroundSize": "",
          "display": "inline-block",
          "height": "96px",
          "width": "96px",
        }
      }
    >
      <span
        className=""
        style={
          Object {
            "display": "inline-block",
            "height": "96px",
            "width": "96px",
          }
        }
      />
    </span>
  </div>
</div>
`;
